mosip-identity-plugin/src/test/java/io/mosip/esignet/plugin/mosipid/service/IdaAuthenticatorImplTest.java

```diff
@@ -143,6 +144,78 @@ public void doKycAuth_withValidDetails_thenPass() throws Exception {
 		Assert.assertEquals(kycAuthResult.getKycToken(), kycAuthResult.getKycToken());
 	}
 
+	@Test
+	public void doKycAuth_withInValidResponseDetails_thenFail() throws Exception {
+		KycAuthDto kycAuthDto = new KycAuthDto();
+		kycAuthDto.setIndividualId("IND1234");
+		kycAuthDto.setTransactionId("TRAN1234");
+		AuthChallenge authChallenge = new AuthChallenge();
+		authChallenge.setAuthFactorType("OTP");
+		authChallenge.setChallenge("111111");
+		List<AuthChallenge> authChallengeList = new ArrayList<>();
+		authChallengeList.add(authChallenge);
+		kycAuthDto.setChallengeList(authChallengeList);
+
+		Mockito.when(mapper.writeValueAsString(Mockito.any())).thenReturn("value");
+
+		IdaKycAuthResponse idaKycAuthResponse = new IdaKycAuthResponse();
+		idaKycAuthResponse.setAuthToken("authToken1234");
+		idaKycAuthResponse.setKycToken("kycToken1234");
+		idaKycAuthResponse.setKycStatus(true);
+
+		IdaResponseWrapper<IdaKycAuthResponse> idaResponseWrapper = new IdaResponseWrapper<>();
+		idaResponseWrapper.setTransactionID("TRAN123");
+		idaResponseWrapper.setVersion("VER1");
+
+		ResponseEntity<IdaResponseWrapper<IdaKycAuthResponse>> responseEntity = new ResponseEntity<IdaResponseWrapper<IdaKycAuthResponse>>(
+				idaResponseWrapper, HttpStatus.OK);
+
+		Mockito.when(restTemplate.exchange(Mockito.<RequestEntity<Void>>any(),
+						Mockito.<ParameterizedTypeReference<IdaResponseWrapper<IdaKycAuthResponse>>>any()))
+				.thenReturn(responseEntity);
+		try{
+			idaAuthenticatorImpl.doKycAuth("relyingId", "clientId", kycAuthDto);
+		}catch (KycAuthException e){
+			Assert.assertEquals(e.getErrorCode(), ErrorConstants.AUTH_FAILED);
+		}
+	}
+
+	@Test
+	public void doKycAuth_withInValidStatusCode_thenFail() throws Exception {

```

Comment:
how is this method different from doKycAuth_withInValidResponseDetails_thenFail  test method??

Reply:
This method is check for status code not equal to 200 while this doKycAuth_withInValidResponseDetails_thenFail() method test the case when we get the status code 200 but get response as null

Reply:
with these two we are covering some extra line of code